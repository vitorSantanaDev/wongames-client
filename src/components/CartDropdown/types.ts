import { GameItemProps } from 'components/GameItem/types'

export type CartDropdownProps = {
  items?: GameItemProps[]
  total?: string
}
