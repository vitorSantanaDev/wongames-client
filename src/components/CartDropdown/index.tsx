import CartList from 'components/CartList'
import Dropdown from 'components/Dropdown'
import CartIcon from 'components/CartIcon'

import { CartDropdownProps } from './types'

import * as S from './styles'

const CartDropdown = ({ items, total }: CartDropdownProps) => (
  <S.Wrapper>
    <Dropdown title={<CartIcon />}>
      <CartList items={items} total={total} hasButton />
    </Dropdown>
  </S.Wrapper>
)

export default CartDropdown
