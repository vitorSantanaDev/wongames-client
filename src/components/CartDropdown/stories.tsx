import { Story, Meta } from '@storybook/react'
import CartDropdown from '.'
import { CartDropdownProps } from './types'

import itemsMock from 'components/CartList/mock'

export default {
  title: 'CartDropdown',
  component: CartDropdown,
  parameters: { backgrounds: { default: 'won-dark' } },
  args: { items: [...itemsMock], total: 'R$ 605,00' }
} as Meta

export const Default: Story<CartDropdownProps> = (args) => (
  <div style={{ maxWidth: '98%', display: 'flex', justifyContent: 'center' }}>
    <CartDropdown {...args} />
  </div>
)

export const Empty: Story = () => (
  <div style={{ maxWidth: '98%', display: 'flex', justifyContent: 'center' }}>
    <CartDropdown />
  </div>
)
