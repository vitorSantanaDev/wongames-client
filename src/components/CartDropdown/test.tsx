import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import CartDropdown from '.'

import itemsMock from 'components/CartList/mock'

const props = { items: itemsMock, total: 'R$ 605,00' }

describe('<CartDropdown />', () => {
  it('should render the <CartIcon /> and its badge', () => {
    const { container } = renderWithTheme(<CartDropdown {...props} />)

    const shoppingCartIcon = screen.getByLabelText(/shopping cart/i)
    expect(shoppingCartIcon).toBeInTheDocument()

    const shoppingCartIconBadge = screen.getByText(itemsMock.length)
    expect(shoppingCartIconBadge).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render dropdown content with cart items and total', () => {
    renderWithTheme(<CartDropdown {...props} />)

    const price = screen.getByText(props.total)
    expect(price).toBeInTheDocument()

    const item = screen.getByText(props.items[0].title)
    expect(item).toBeInTheDocument()
  })
})
