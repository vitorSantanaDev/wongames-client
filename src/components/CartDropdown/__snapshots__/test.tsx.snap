// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartDropdown /> should render the <CartIcon /> and its badge 1`] = `
.c18 {
  border: none;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FAFAFA;
  padding: 0.8rem;
  border-radius: 0.4rem;
  background: linear-gradient(180deg,#ff605f 0%,#f062c0 50%);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c18:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c10 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c12 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c14 {
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  line-height: 1.4rem;
  margin-bottom: 0.8rem;
}

.c15 {
  color: #FAFAFA;
  font-weight: 600;
  padding: 0.2rem 0.8rem;
  font-size: 1.4rem;
  border-radius: 0.4rem;
  background-color: #3CD3C1;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #FAFAFA;
}

.c16 {
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  background-color: #eee;
}

.c17 {
  color: #F231A5;
}

.c2 {
  cursor: pointer;
  color: #FAFAFA;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2.4rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FAFAFA;
  color: #030517;
  margin-top: 2.4rem;
  position: absolute;
  right: 0;
}

.c7::before {
  content: '';
  position: absolute;
  border-right: 1.2rem solid transparent;
  border-left: 1.2rem solid transparent;
  border-bottom: 1.2rem solid #FAFAFA;
  top: -1.2rem;
  right: 2.4rem;
}

.c1 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c1 .c6 {
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-2rem);
  -ms-transform: translateY(-2rem);
  transform: translateY(-2rem);
}

.c3 {
  width: 2.4rem;
  height: 2.4rem;
  color: #FAFAFA;
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FAFAFA;
  background-color: #3CD3C1;
  font-size: 1rem;
  border-radius: 50%;
  width: 1.6rem;
  height: 1.6rem;
  position: absolute;
  top: -0.4rem;
  right: -0.4rem;
}

.c0 .c8 {
  width: 56rem;
}

@media (min-width:768px) {
  .c10 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c12 {
    width: 15rem;
    height: 7rem;
    margin-right: 1.6rem;
  }
}

@media (min-width:768px) {
  .c14 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

@media (min-width:768px) {
  .c16 {
    padding: 2.4rem;
    font-size: 1.6rem;
  }
}

<div
  class="c0"
>
  <main
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          aria-label="cart items"
          class="c4"
        >
          2
        </div>
        <svg
          aria-hidden="true"
          aria-label="shopping cart"
          class="c5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0 0 20.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="c6 c7"
    >
      <div
        class="c8 c9"
      >
        <div
          class="c10"
        >
          <div
            class="c11"
          >
            <div
              class="c12"
            >
              <img
                alt="Hoziron II"
                src="https://wallpaperaccess.com/full/3149843.jpg"
              />
            </div>
            <div
              class="c13"
            >
              <h3
                class="c14"
              >
                Hoziron II
              </h3>
              <p
                class="c15"
              >
                R$ 350,00
              </p>
            </div>
          </div>
        </div>
        <div
          class="c10"
        >
          <div
            class="c11"
          >
            <div
              class="c12"
            >
              <img
                alt="Halo 5"
                src="https://wallpapers-hub.art/wallpaper-images/263763.jpg"
              />
            </div>
            <div
              class="c13"
            >
              <h3
                class="c14"
              >
                Halo 5
              </h3>
              <p
                class="c15"
              >
                R$ 255,00
              </p>
            </div>
          </div>
        </div>
        <div
          class="c16"
        >
          <span
            class="c17"
          >
            R$ 605,00
          </span>
          <a
            class="c18"
            href="/cart"
          >
            <span>
              Buy it now
            </span>
          </a>
        </div>
      </div>
    </div>
  </main>
</div>
`
